core/execution/future/market.go

```diff
@@ -1821,11 +1821,13 @@ func (m *Market) triggerStopOrders(
 	if m.lastTradedPrice == nil {
 		return nil
 	}
-
 	lastTradedPrice := m.priceToMarketPrecision(m.getLastTradedPrice())
 	triggered, cancelled := m.stopOrders.PriceUpdated(lastTradedPrice)
 
-	if len(triggered) <= 0 {
+	// See if there are any linked orders that are the wrong direction
+	cancelled = append(cancelled, m.stopOrders.CheckDirection(m.position)...)
```

Comment:
Why are we doing this here?

Reply:
I'm checking to see if the position has flipped sign every time we evaluate if we have a stop order to process. If felt right to process them both at the same place. Let me know if you have a better place for it.